### PR TITLE
Authorize teachers who enroll in PD for CSD and CSP

### PR DESCRIPTION
**Pull request by mehalshah**
Also fix the apostrophe bug in free responses for facilitator specific answers
Also remove CSP from the list of auto-enrolled courses

### CODE REVIEW

**Comment by aoby** [comment]
That's right @clareconstantine on both accounts. 

The concern was that a pre-enrollment may not have a user association. In those cases the user is only saved on attendance, and that's what we wanted to capture. This covers it by updating every time in the `after_save` callback. 

To be more clear, we could add a test for that specific scenario:
 1. enroll with email-only, no user account
 2. attend and update enrollment with user account
 3. assert the new user has been verified

**Comment by mehalshah** [comment]
Also did https://trello.com/c/RsjiT7OT/654-teachers-who-enroll-in-a-csp-workshop-are-automatically-co-enrolled-in-the-plc-course-remove-this-automatic-co-enrollment-so-plc here

**Comment by mehalshah** [comment]
https://trello.com/c/69kfFZLW/634-malformatting-of-apostrophes-in-survey-responses

**Comment by mehalshah** [comment]
https://trello.com/c/NzfCF7FY/653-automatically-verify-all-teacher-accounts-who-enroll-in-any-csd-p-workshop

File: dashboard/test/models/pd/enrollment_test.rb [hunk 1]
```diff
@@ -490,4 +490,12 @@ class Pd::EnrollmentTest < ActiveSupport::TestCase
...
     refute teacher.permission? UserPermission::AUTHORIZED_TEACHER
   end
+
+  test 'Updating existing enrollment sets permission' do
+    teacher = create :teacher
+    workshop = create :pd_workshop, course: Pd::SharedWorkshopConstants::COURSE_CSD, num_sessions: 1
+    create :pd_enrollment, workshop: workshop, user: nil, email: teacher.email
+
+    assert teacher.permission? UserPermission::AUTHORIZED_TEACHER
+  end
```
Comment (aoby) [inline dashboard/test/models/pd/enrollment_test.rb]: I expect this to fail. It needs to first update the enrollment from line 497 to associated it with the user. Ideally, first refute that the (unassociated) user is not verified.

**Pull request merged**